test/testvm.py

```diff
@@ -971,15 +976,34 @@ def wait_reboot(self):
             raise Failure("system didn't reboot properly")
         self.wait_user_login()
 
-    def wait_boot(self, wait_for_running_timeout = 120):
+    def wait_boot(self, wait_for_running_timeout = 120, allow_one_reboot=False):
         # we should check for selinux relabeling in progress here
         if not self.event_handler.wait_for_running(self._domain, timeout_sec=wait_for_running_timeout ):
             raise Failure("Machine %s didn't start." % (self.address))
 
-        if not Machine.wait_ssh(self, get_new_address = lambda: self._ip_from_mac(self.macaddr, timeout_sec=5)):
-            self._diagnose_no_address()
-            raise Failure("Unable to reach machine %s via ssh." % (self.address))
-        self.wait_user_login()
+        # if we allow a reboot, the connection to test for a finished boot may be interrupted
+        # by the reboot, causing the exception
+        try:
+            start_time = time.time()
+            connected = False
+            while (time.time() - start_time) < wait_for_running_timeout:
+                if Machine.wait_ssh(self, timeout_sec=15, get_new_address = lambda: self._ip_from_mac(self.macaddr, timeout_sec=3)):
+                    connected = True
+                    break
+                if self.event_handler.has_rebooted(self._domain):
```

Comment:
@stefwalter Your comment made me realize that this if should also include `allow_one_reboot`